### PR TITLE
Add Second Timestamp Checker

### PR DESCRIPTION
**Pull request by harrybiddle**
This PR introduces 'FailEvents' which are of type `:fail`, with error strings. In order to be able to pass the error strings back to Jepsen, I've removed the code that converts all strings into keywords when preparing the Java objects for Jepsen.

We also introduce an `ImpossibleServerChecker`. When a read is acknowledged, we take the value `t`, find the invocation time of the read, and check that `t` is bigger than the last read acknowledged before this time.

The check does **not** confirm that all acknowledged or failed reads are paired with invokes.

This PR should also include any outstanding changes from #1278.

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/1295)

### CODE REVIEW

File: atlasdb-timelock-server/build.gradle [hunk 1]
```diff
@@ -39,3 +39,9 @@ distribution {
     mainClass 'com.palantir.atlasdb.timelock.TimeLockServer'
     args 'server', 'var/conf/timelock.yml'
 }
+
+distribution {
```
Comment (SerialVelocity) [inline atlasdb-timelock-server/build.gradle]: ? Copy paste error?

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/JepsenHistoryChecker.java [hunk 2]
```diff
@@ -36,7 +36,7 @@ public JepsenHistoryChecker(Checker... checkers) {
     }
 
     public static JepsenHistoryChecker createWithStandardCheckers() {
-        return new JepsenHistoryChecker(new MonotonicChecker());
+        return new JepsenHistoryChecker(new MonotonicChecker(), new ImpossibleServerChecker());
```
Comment (SerialVelocity) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/JepsenHistoryChecker.java]: Did you forget to commit this file?

File: atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/events/EventTest.java [hunk 3]
```diff
@@ -92,6 +92,44 @@ public void canDeserialiseOkRead() {
     }
 
     @Test
+    public void canDeserialiseFailEventWithStacktrace() {
+        final String exceptionString = new RuntimeException("Error").toString();
```
Comment (SerialVelocity) [inline atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/events/EventTest.java]: Shouldn't need to add `final` anywhere. Java 8+ automatically adds them where necessary

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/JepsenHistoryChecker.java (hunk 2 above)
Comment (harrybiddle) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/JepsenHistoryChecker.java]: Ugh, this and three others, d'oh. All added

File: atlasdb-timelock-server/build.gradle (hunk 1 above)
Comment (harrybiddle) [inline atlasdb-timelock-server/build.gradle]: Yup! Removed

File: atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/ImpossibleServerChecker.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+import com.palantir.atlasdb.jepsen.events.Checker;
+import com.palantir.atlasdb.jepsen.events.Event;
+import com.palantir.atlasdb.jepsen.events.EventVisitor;
+import com.palantir.atlasdb.jepsen.events.FailEvent;
+import com.palantir.atlasdb.jepsen.events.ImmutableOkEvent;
+import com.palantir.atlasdb.jepsen.events.InfoEvent;
+import com.palantir.atlasdb.jepsen.events.InvokeEvent;
+import com.palantir.atlasdb.jepsen.events.OkEvent;
+
+public class ImpossibleServerChecker implements Checker {
```
Comment (SerialVelocity) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/ImpossibleServerChecker.java]: This name gives me no idea what it does. Does it always fail? xD

Comment (harrybiddle) [inline atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/ImpossibleServerChecker.java]: Changed to `NonOverlappingReadsMonotonicChecker`

**Comment by harrybiddle** [comment]
Ready for re-review. Only change is that I've renamed `ImpossibleServerChecker` -> `NonOverlappingReadsMonotonicChecker`.

**Pull request merged**